dev-docs/RFCs/v1.0/react-map-gl-draw.md

```diff
@@ -0,0 +1,156 @@
+# RFC: react-map-gl-draw
+
+## Background
+
+react-map-gl currently does not support drawing functions. However, we have got a couple of [users](https://github.com/uber/react-map-gl/issues/725) interested in this capability. Also it is one of P0 features on Kepler.gl 2019 [roadmap](https://github.com/uber/kepler.gl/wiki/Kepler.gl-2019-Roadmap#allow-drawing-on-map-to-create-paths-and-polygons--). 
+
+Although [Mapbox/mapbox-gl-draw](https://github.com/mapbox/mapbox-gl-draw) provides quite nice drawing and editing features, because of its manipulating internal states, it cannot work well with React / Redux  framework and therefore cannot be integrated with `react-map-gl`.
+[Nebula.gl](http://neb.gl) provides geo editing functionality, however it heavily depends on `deck.gl`.
+
+## Proposal
+
+`react-map-gl-draw` is a react based geo editing library, providing functions like the following.
+
+### Options 
+- `MapContextProvider` (React.Component, required) - A [Context.Provider](https://reactjs.org/docs/context.html#contextprovider) component. If supplied, will be rendered as the ancestor to all children. The passed through context contains the following values:
+  - `viewport` ([Viewport](/docs/api-reference/viewport.md)) - the current viewport
+  - `eventManager` ([EventManager](https://github.com/uber-web/mjolnir.js/blob/master/docs/api-reference/event-manager.md))
+
+- `mode` (String, Optional) - `react-map-gl` is stateless, user has complete control of the `mode`.
+  - `EditorModes.READ_ONLY` - Not interactive. This is the default mode.
+  - `EditorModes.SELECT_FEATURE` - Lets you select, delete, and drag features.
+  - `EditorModes.EDIT_VERTEX` - Lets you select, delete, and drag vertices; and drag features.
+  - `EditorModes.DRAW_PATH` - Lets you draw a LineString feature.
+  - `EditorModes.DRAW_POLYGON` - Lets you draw a Polygon feature.
+  - `EditorModes.DRAW_POINT` - Lets you draw a Point feature.
+  - `EditorModes.DRAW_RECTANGLE` - Lets you draw a Rectangle feature.
+
+- `getFeatureStyle` (Function, Optional) : Object - A function to style features, function parameters are 
```

Comment:
isn't `getFeatureStyle` should be for `path` element, and `getVertexStyle` should be for `circle` and `rect`?

Reply:
Updated the APIs. `getVertexStyle` is kind of exposing the implementation details. So decided to use `getEditHandleStyle`, which will be applied to the feature being edited.